src/term/mod.rs

```diff
@@ -55,39 +77,66 @@ impl<'a> RenderableCellsIter<'a> {
     fn new<'b>(
         grid: &'b mut Grid<Cell>,
         cursor: &'b Point,
+        cursor_style: CursorStyle,
         mode: TermMode,
         selection: &Selection,
         custom_cursor_colors: bool,
     ) -> RenderableCellsIter<'b> {
         let selection = selection.span()
             .map(|span| span.to_range(grid.num_cols()));
 
+        let cell = IndexedCell {
+            line: cursor.line,
+            column: cursor.col,
+            inner: grid[cursor]
+        };
+
+        let cursor_cell = match cursor_style.0 {
+            ansi::CursorStyle::Block => {
+                CursorCell::Replacement(cell)
+            }
+            ansi::CursorStyle::Underline => {
+                CursorCell::Extra(cell)
+            }
+            ansi::CursorStyle::Beam => {
+                CursorCell::Extra(cell)
+            }
+        };
+
         RenderableCellsIter {
             grid: grid,
             cursor: cursor,
+            cursor_style: cursor_style,
```

Comment:
Instead of adding the `cursor_style` property to the `RendesableCellsIter`, I think we can get away with just updating the character in `CursorCell::Extra` to be the appropriate thing (bar/underline).